internal/ackhandler/sent_packet_handler.go

```diff
@@ -294,40 +309,98 @@ func (h *sentPacketHandler) detectLostPackets(now time.Time) {
 	})
 
 	for _, p := range lostPackets {
-		h.queuePacketForRetransmission(p)
+		if err := h.queuePacketForRetransmission(p); err != nil {
+			return err
+		}
 		h.congestion.OnPacketLost(p.PacketNumber, p.Length, h.bytesInFlight)
 	}
+	return nil
 }
 
-func (h *sentPacketHandler) OnAlarm() {
+func (h *sentPacketHandler) OnAlarm() error {
+	utils.Debugf("onAlarm")
 	now := time.Now()
 
 	// TODO(#497): TLP
+	var err error
 	if !h.handshakeComplete {
-		h.queueHandshakePacketsForRetransmission()
+		err = h.queueHandshakePacketsForRetransmission()
 		h.handshakeCount++
 	} else if !h.lossTime.IsZero() {
 		// Early retransmit or time loss detection
-		h.detectLostPackets(now)
+		err = h.detectLostPackets(now)
 	} else {
 		// RTO
-		h.retransmitOldestTwoPackets()
+		err = h.queueRTOs()
 		h.rtoCount++
 	}
 
 	h.updateLossDetectionAlarm(now)
```

Comment:
It's weird that we continue to do work after an error was returned – but I guess it doesn't really matter in this case.

Reply:
I agree, and it's easy to fix. Just one additional `if`.